src/scenes/lobbyscreen.ts

```diff
@@ -0,0 +1,148 @@
+        const buttonText = this.add.text(0, 0, "Click me", { 
+            fontFamily: "Arial", 
+            fontSize: "20px", 
+            color: "#FFFFFF",
+        });
+
+        Phaser.Display.Align.In.Center(buttonText, demoButton);
+
+        // Make the text interactive
+        demoButton.setInteractive();
+        // Add a pointerdown event listener to the text
+        demoButton.on("pointerdown", () => {
+            log("click me!");
+            // Add your custom functionality here
+        });
+        demoButton.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        demoButton.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+        const startButton = this.add.image(150, 540, "testButton");
+        startButton.setInteractive();
+        startButton.on("pointerdown", () => {
+            log("Click brown button"); // Call a custom function when the button is clicked
+        });
+        startButton.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        startButton.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+        const createText = this.add.text(580, 540, "Start", { 
+            color: "#ffffff", 
+            fontSize: "50px",
+        });
+        createText.setInteractive();
+        createText.on("pointerdown", () => {
+            log("Click on start text");
+        });
+        createText.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        createText.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+    }
+
+    private playerBackground(ypos: integer): void{
+        // Placeholder for the background on player display
+        this.graphics = this.add.graphics();
+        this.graphics.lineStyle(1, 0x000000, 1);
+        this.graphics.fillStyle(0x5C4033, 1.0);
+        // xpos, ypos, horizontal length, vertical length, corner radius
+        this.graphics.fillRoundedRect(100, ypos, 600, 27, 5);
+        this.graphics.strokeRoundedRect(100, ypos, 600, 27, 5);
+    }
+
+    public override update(): void {
+        for (let i=0; i<this.players.length; i++){
+            let ypos = 150 + i*30;
+            let playername = this.players[i]
+            let fontstyle = "normal";
+            if (this.players[i] === this.me[1]){
+                playername = "You are: " + playername + ", Role: TBD";
+                fontstyle = "oblique";
+            }
+            this.playerBackground(ypos);
+            this.add.text(100, ypos, playername, { 
+                fontFamily: "Arial Black",
+                fontSize: 20,
+                color: "#e1e1e1",
+                stroke: "#000000",
+                strokeThickness: 3,
+                fontStyle: fontstyle,                
+            });
+        }
+    }
```

Comment:
Instead of update frame driven please make it event driven. so that the lobby information doesnt update each frame but update only when the inforamtion changes and then i think you can also leave away the private me and players field

Reply:
Looks much better thanks!